node/bft/src/bft.rs

```diff
@@ -855,6 +862,30 @@ impl<N: Network> BFT<N> {
                 callback.send(result).ok();
             }
         });
+
+        // Process the request to commit the leader certificate. 
+        let self_ = self.clone();
+        self.spawn(async move {
+            while let Some((certificate, callback)) = rx_commit_bft.recv().await {
+                // Update the DAG with the certificate.
+                let result = self_.commit_leader_certificate::<true, true>(certificate).await;
+                // Send the callback **after** updating the DAG.
+                // Note: We must await the DAG update before proceeding.
+                callback.send(result).ok();
+            }
+        });
+
+        // Process the request to check if the batch certificate was recently committed.
+        let self_ = self.clone();
+        self.spawn(async move {
+            while let Some(((round, certificate_id), callback)) = rx_is_recently_committed.recv().await {
+                // Check if the certificate was recently committed.
+                let is_committed = self_.dag.read().is_recently_committed(round, certificate_id);
```

Comment:
seems like a lot of hassle just to perform this one check in its own task with a dedicated channel, especially with both the `BFT` and `Dag` being clonable :thinking: